### PR TITLE
Dartino on stm32 from linux

### PR DESCRIPTION
**Pull request by danrubel**
@devoncarew 
refactor launching external processesubject
support Dartino on STM32 from Linuxubject
save dirty editors before build appubject
cleanup importsubject
address comments from https://github.com/dartino/atom-dartino/pull/2ubject

### CODE REVIEW

File: lib/dartino.dart [hunk 1]
```diff
@@ -10,10 +10,9 @@ import 'package:atom/atom.dart';
 import 'package:atom/node/process.dart';
 import 'package:atom/utils/disposable.dart';
 import 'package:atom_dartino/sdk/sdk.dart';
+import 'package:atom_dartino/usb.dart';
```
Comment (devoncarew) [inline lib/dartino.dart:13]: I use relative path references for code in lib/ that also refers to lib/ code. Don't think that's a 1st class convention, but something I do.

File: lib/sdk/dartino_sdk.dart [hunk 2]
```diff
@@ -13,6 +14,7 @@ import 'package:atom_dartino/sdk/sdk.dart';
 import 'package:atom_dartino/sdk/sdk_util.dart';
 
 import '../dartino.dart' show pluginId;
+import 'package:atom_dartino/proc.dart';
```
Comment (devoncarew) [inline lib/sdk/dartino_sdk.dart:17]: may want to sort these imports :)

**Comment by devoncarew** [comment]
lgtm

File: lib/dartino.dart (hunk 1 above)
Comment (danrubel) [inline lib/dartino.dart:13]: Done as part of a later commit https://github.com/dartino/atom-dartino/commit/723e50a2a1e900e3a7e41fa5d9e29a6faa955e70

File: lib/sdk/dartino_sdk.dart (hunk 2 above)
Comment (danrubel) [inline lib/sdk/dartino_sdk.dart:17]: Done as part of a later commit https://github.com/dartino/atom-dartino/commit/723e50a2a1e900e3a7e41fa5d9e29a6faa955e70

**Pull request merged**